airflow/contrib/operators/sagemaker_training_operator.py

```diff
@@ -0,0 +1,101 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.contrib.operators.sagemaker_base_operator import SageMakerBaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import AirflowException
+
+
+class SageMakerTrainingOperator(SageMakerBaseOperator):
+    """
+    Initiate a SageMaker training job
+    This operator returns The ARN of the training job created in Amazon SageMaker
+
+    :param config: The configuration necessary to start a training job (templated)
+    :type config: dict
+    :param aws_conn_id: The AWS connection ID to use.
+    :type aws_conn_id: str
+    :param wait_for_completion: if the operator should block until training job finishes
+    :type wait_for_completion: bool
+    :param print_log: if the operator should print the cloudwatch log during training
+    :type print_log: bool
+    :param check_interval: if wait is set to be true, this is the time interval
+        in seconds which the operator will check the status of the training job
+    :type check_interval: int
+    :param max_ingestion_time: if wait is set to be true, the operator will fail
+        if the training job hasn't finish within the max_ingestion_time in seconds
+        (Caution: be careful to set this parameters because training can take very long)
+        Setting it to None implies no timeout.
+    :type max_ingestion_time: int
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 aws_conn_id='aws_default',
+                 wait_for_completion=True,
+                 print_log=True,
+                 check_interval=30,
+                 max_ingestion_time=None,
+                 *args, **kwargs):
+        super(SageMakerTrainingOperator, self).__init__(config=config,
+                                                        aws_conn_id=aws_conn_id,
+                                                        *args, **kwargs)
+
+        self.wait_for_completion = wait_for_completion
+        self.print_log = print_log
+        self.check_interval = check_interval
+        self.max_ingestion_time = max_ingestion_time
+        self.create_integer_fields()
+
+    def create_integer_fields(self):
+        self.integer_fields = [
+            ['ResourceConfig', 'InstanceCount'],
+            ['ResourceConfig', 'VolumeSizeInGB'],
+            ['StoppingCondition', 'MaxRuntimeInSeconds']
+        ]
+
+    def expand_role(self):
+        if 'RoleArn' in self.config:
+            self.config['RoleArn'] = self.hook.expand_role(self.config['RoleArn'])
+
+    def execute(self, context):
+        self.preprocess_config()
+
+        self.log.info(
+            'Creating SageMaker Training Job %s.'
+            % self.config['TrainingJobName']
```

Comment:
Don't need the `%`, pass extra argument to log.info instead:

```
        self.log.info('Creating SageMaker Training Job %s.', self.config['TrainingJobName'])
```

Reply:
This applies to all/most of the log statements from in the operators too.

Reply:
Updated in all operators.